### PR TITLE
chassis and blades serialze each other in their network graphs

### PR DESCRIPTION
**Pull request by benauriemma**

### CODE REVIEW

File: rackcity/api/serializers/asset_serializers.py [hunk 1]
```diff
@@ -530,11 +531,43 @@ def get_neighbor_assets(hostname, id, nodes, edges, change_plan=None):
...
         return nodes, edges
     except ObjectDoesNotExist:
         return
 
 
+def add_blades_to_graph(id, nodes, edges, change_plan=None):
+    if change_plan:
+        blades = AssetCP.objects.filter(chassis=id, change_plan=change_plan)
+    else:
+        blades = Asset.objects.filter(chassis=id)
```
Comment (clairecfu) [inline rackcity/api/serializers/asset_serializers.py:545]: even if youre on a change plan, you still need to get the blades for a chassis that might not be on the change plan.

File: rackcity/api/serializers/asset_serializers.py [hunk 2]
```diff
@@ -546,7 +546,12 @@ def get_blades_in_chassis_cp(asset_cp):
 def generate_network_graph(asset):
     try:
         nodes = []
-        nodes.append({"id": asset.id, "label": asset.hostname})
+        if hasattr(asset, "related_asset") and asset.related_asset:
+
+            route_id = asset.related_asset.id
+        else:
+            route_id = asset.id
+        nodes.append({"id": asset.id, "route_id": route_id, "label": asset.hostname})
```
Comment (clairecfu) [inline rackcity/api/serializers/asset_serializers.py:554]: i do this everywhere -> that way route_id is always the id of the live asset if it exists

File: rackcity/api/serializers/asset_serializers.py [hunk 3]
```diff
@@ -571,31 +577,85 @@ def generate_network_graph(asset):
...
+        edges.append({"from": chassis_id, "to": blade.id})
+    return nodes, edges
+
+
+def add_chassis_to_graph(blade_id, nodes, edges, change_plan=None):
+    if change_plan:
+        chassis = AssetCP.objects.get(id=blade_id).chassis
+    else:
+        chassis = Asset.objects.get(id=blade_id).chassis
+    if not chassis:
```
Comment (clairecfu) [inline rackcity/api/serializers/asset_serializers.py:642]: the only thing i changed was adding change_plan here

File: src/components/elementView/detailedView/assetView/assetView.tsx [hunk 4]
```diff
@@ -434,18 +434,18 @@ export class AssetView extends React.PureComponent<
...
-                <tr>
-                  <th>Network Port</th>
-                  <th>Mac Address</th>
-                  <th>Destination Asset</th>
-                  <th>Destination Port</th>
-                </tr>
+              {this.state.asset.model &&
+              this.state.asset.model.network_ports &&
+              this.state.asset.model.network_ports.length !== 0 ? (
+                              <table className="bp3-html-table bp3-html-table-bordered bp3-html-table-striped">
```
Comment (clairecfu) [inline src/components/elementView/detailedView/assetView/assetView.tsx:441]: don't show network table headers if no network ports

File: src/components/elementView/detailedView/assetView/chassisView.tsx [hunk 5]
```diff
@@ -12,7 +12,7 @@ interface ChassisViewProps {
 class ChassisView extends React.PureComponent<
   RouteComponentProps & ChassisViewProps
 > {
-  numSlots = 12;
+  numSlots = 14;
```
Comment (clairecfu) [inline src/components/elementView/detailedView/assetView/chassisView.tsx:15]: 14!!!!

File: rackcity/api/serializers/asset_serializers.py [hunk 6]
```diff
@@ -319,7 +319,7 @@ class RecursiveAssetCPSerializer(serializers.ModelSerializer):
 
     model = ITModelSerializer()
     rack = RackSerializer()
-    chassis = ChassisSerializer()
+    chassis = ChassisCPSerializer()
```
Comment (clairecfu) [inline rackcity/api/serializers/asset_serializers.py:322]: this is a bug fix for chassis detail view

File: src/components/elementView/detailedView/assetView/assetView.scss [hunk 7]
```diff
@@ -34,17 +32,16 @@
...
+
 
   display: block;
 }
 .graph-container svg {
   width: 100% !important;
 }
 .network-connections {
-  display: flex;
+  display: block;
```
Comment (benauriemma) [inline src/components/elementView/detailedView/assetView/assetView.scss:43]: Does this make it so the graph isn't huge when there's no table?

Comment (clairecfu) [reply src/components/elementView/detailedView/assetView/assetView.scss:43]: i made the graph show up below the table so its big but the same size between assets

**Pull request merged**